server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -37,13 +37,31 @@ public class EsExecutors {
     /**
      * Setting to manually control the number of allocated processors. This setting is used to adjust thread pool sizes per node. The
      * default value is {@link Runtime#availableProcessors()} but should be manually controlled if not all processors on the machine are
-     * available to Elasticsearch (e.g., because of CPU limits).
+     * available to Elasticsearch (e.g., because of CPU limits). Note that this setting accepts floating point processors, so it should not
+     * be used directly, use always {@link EsExecutors#allocatedProcessors(Settings)} to obtain a rounded number of processors.
```

Comment:
```suggestion
     * available to Elasticsearch (e.g., because of CPU limits). Note that this setting accepts floating point processors. If a rounded number is
     * needed, always use {@link EsExecutors#allocatedProcessors(Settings)}.
```